osmoutils/accum/accum.go

```diff
@@ -83,16 +88,49 @@ func setAccumulator(accum_store store.KVStore, accum AccumulatorObject, amt sdk.
 }
 
 // TODO: consider making this increment the accumulator's value instead of overwriting it
-func (accum AccumulatorObject) UpdateAccumulator(amt sdk.Dec) {
+func (accum AccumulatorObject) UpdateAccumulator(amt sdk.DecCoins) {
 	setAccumulator(accum.Store, accum, amt)
 }
 
-// func (accum AccumulatorObject) NewPosition(addr, num_units, positionOptions) error
+// NewPosition creates a new position for the given address, with the given number of units
+// It takes a snapshot of the current accumulator value, and sets the position's initial value to that
+// The position is initialized with empty unclaimed rewards
+func (accum AccumulatorObject) NewPosition(addr sdk.AccAddress, numShareUnits sdk.Dec, options PositionOptions) {
+	position := Record{
+		NumShares:        numShareUnits,
+		InitAccumValue:   accum.Value,
+		UnclaimedRewards: sdk.NewDecCoins(),
+	}
+	osmoutils.MustSet(accum.Store, addr, &position)
```

Comment:
Is it okay to use the address directly as the key? Should we prefix it with "accumulator-" or something to avoid conflicts with other things that might store directly into the address key?

Reply:
Please see: https://github.com/osmosis-labs/osmosis/pull/3855#discussion_r1056886112

and the `prefix.go` file for relevant updates